src/services/tx_confirmation.rs

```diff
@@ -0,0 +1,37 @@
+use crate::config::base_transaction_service_url;
+use crate::utils::cache::Cache;
+use crate::utils::context::Context;
+use crate::utils::errors::{ApiError, ApiResult};
+use std::collections::HashMap;
+
+pub fn submit_confirmation(
+    context: &Context,
+    safe_tx_hash: &str,
+    signature: &str,
+) -> ApiResult<()> {
+    context
+        .cache()
+        .invalidate_pattern(&format!("*{}*", &safe_tx_hash));
+
+    let url = format!(
+        "{}/v1/multisig-transactions/{}/confirmations/",
+        base_transaction_service_url(),
+        &safe_tx_hash
+    );
+    let mut json = HashMap::new();
+    json.insert("signature", signature);
+
+    let result = context.client().post(&url).json(&json).send()?;
+
+    if result.status().is_success() {
+        Ok(())
+    } else {
+        Err(ApiError::from_backend_error(
+            result.status().as_u16(),
```

Comment:
Would it be helpful to add a helper to `ApiError` that takes the result?

Reply:
Ah, that's not really a `Result` but a `Response` from reqwest. The var name is a bit confusing. But I think it's a good idea to have a default mapper like that